modules/littlefs/zephyr_lfs_config.h

```diff
@@ -178,26 +167,15 @@ static inline uint32_t lfs_tole32(uint32_t a)
 /* Convert between 32-bit big-endian and native order */
 static inline uint32_t lfs_frombe32(uint32_t a)
 {
-#if !defined(LFS_NO_INTRINSICS) && ( \
-	(defined(BYTE_ORDER) && defined(ORDER_LITTLE_ENDIAN) &&         \
-	BYTE_ORDER == ORDER_LITTLE_ENDIAN) ||                           \
-	(defined(__BYTE_ORDER) && defined(__ORDER_LITTLE_ENDIAN) &&     \
-	__BYTE_ORDER == __ORDER_LITTLE_ENDIAN) ||                       \
-	(defined(__BYTE_ORDER__) && defined(__ORDER_LITTLE_ENDIAN__) && \
-	__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__))
-	return __builtin_bswap32(a);
-#elif (defined(BYTE_ORDER) && defined(ORDER_BIG_ENDIAN) &&           \
-	BYTE_ORDER == ORDER_BIG_ENDIAN) ||                           \
-	(defined(__BYTE_ORDER) && defined(__ORDER_BIG_ENDIAN) &&     \
-	__BYTE_ORDER == __ORDER_BIG_ENDIAN) ||                       \
-	(defined(__BYTE_ORDER__) && defined(__ORDER_BIG_ENDIAN__) && \
-	__BYTE_ORDER__ == __ORDER_BIG_ENDIAN__)
+#if defined(CONFIG_BIG_ENDIAN)
 	return a;
+#elif !defined(LFS_NO_INTRINSICS)
+	return __builtin_bswap32(a);
 #else
-	return (((uint8_t *)&a)[0] << 24) |
-	       (((uint8_t *)&a)[1] << 16) |
-	       (((uint8_t *)&a)[2] <<  8) |
-	       (((uint8_t *)&a)[3] <<  0);
+	return (((uint8_t *)&a)[0] <<  0) |
+	       (((uint8_t *)&a)[1] <<  8) |
+	       (((uint8_t *)&a)[2] << 16) |
+	       (((uint8_t *)&a)[3] << 24);
```

Comment:
hmm this is also being changed, was it wrong?

Reply:
how strange, no. This is an oversight. Thanks!